crates/rspack_plugin_javascript/src/visitors/dependency/common_js_import_dependency_scanner.rs

```diff
@@ -401,13 +465,34 @@ impl<'a> Visit for CommonJsImportDependencyScanner<'a> {
     }
   }
 
+  fn visit_program(&mut self, program: &Program) {
+    self.is_strict = match program {
+      Program::Module(_) => true,
+      Program::Script(s) => is_strict(&s.body),
+    };
+    program.visit_children_with(self);
+  }
+
   fn visit_block_stmt(&mut self, n: &BlockStmt) {
     let old_in_strict = self.is_strict;
 
-    self.is_strict = is_strict(n);
+    self.is_strict = is_strict(&n.stmts);
```

Comment:
`blockStmts` should not affect the "use strict" within it, for example this case should work:

```rs
if (true) {
  "use strict"
  let = 1
}
```